app/src/main/java/cm/aptoide/pt/home/BottomNavigationNavigator.java

```diff
@@ -0,0 +1,153 @@
+
+  private final static String EVENT_ACTION =
+      "https://ws75.aptoide.com/api/7/getStoreWidgets/store_id=15/context=stores";
+  private final FragmentNavigator fragmentNavigator;
+  private final String defaultStoreName;
+  private final BottomNavigationAnalytics bottomNavigationAnalytics;
+  private final SearchAnalytics searchAnalytics;
+  private ArrayList<Integer> bottomNavigationItens;
+
+  public BottomNavigationNavigator(FragmentNavigator fragmentNavigator, String defaultStoreName,
+      BottomNavigationAnalytics bottomNavigationAnalytics, SearchAnalytics searchAnalytics) {
+    this.defaultStoreName = defaultStoreName;
+    this.bottomNavigationAnalytics = bottomNavigationAnalytics;
+    this.searchAnalytics = searchAnalytics;
+    bottomNavigationItens = new ArrayList<>();
+    this.fragmentNavigator = fragmentNavigator;
+  }
+
+  public void navigateToBottomNavigationItem(int bottomNavigationPosition) {
+    switch (bottomNavigationPosition) {
+      case 0:
+        bottomNavigationAnalytics.sendNavigateToHomeClickEvent();
+        BottomHomeFragment bottomHomeFragment = new BottomHomeFragment();
+        navigateToHome(bottomHomeFragment);
+        break;
+      case 1:
+        bottomNavigationAnalytics.sendNavigateToSearchClickEvent();
+        searchAnalytics.searchStart(SearchSource.BOTTOM_NAVIGATION, true);
+        SearchResultFragment searchResultFragment =
+            SearchResultFragment.newInstance(defaultStoreName, true);
+        navigateToSearch(searchResultFragment);
+        break;
+      case 2:
+        bottomNavigationAnalytics.sendNavigateToStoresClickEvent();
+        MyStoresFragment myStoresFragment =
+            MyStoresFragment.newInstance(getStoreEvent(), "default", "stores", StoreContext.home);
+        navigateToStore(myStoresFragment);
+        break;
+      case 3:
+        bottomNavigationAnalytics.sendNavigateToAppsClickEvent();
+        AppsFragment appsFragment = new AppsFragment();
+        navigateToApps(appsFragment);
+        break;
+    }
+  }
+
+  public void navigateToHome(BottomHomeFragment bottomHomeFragment) {
+    navigateToSelectedFragment(0, bottomHomeFragment);
+  }
+
+  public void navigateToSearch(SearchResultFragment searchResultFragment) {
+    navigateToSelectedFragment(1, searchResultFragment);
+  }
+
+  public void navigateToStore(MyStoresFragment myStoresFragment) {
+    navigateToSelectedFragment(2, myStoresFragment);
+  }
+
+  public void navigateToApps(AppsFragment appsFragment) {
+    navigateToSelectedFragment(3, appsFragment);
+  }
+
+  public void navigateBack() {
+    int bottomNavigationPosition = bottomNavigationItens.get(1);
+    bottomNavigationItens.remove(0);
+    navigateBackToBottomNavigationItem(bottomNavigationPosition);
+  }
+
+  public boolean canNavigateBack() {
+    return bottomNavigationItens.size() > 1;
+  }
+
+  private void checkAndReplaceItem(int newItem) {
+    for (int i = 0; i < bottomNavigationItens.size(); i++) {
```

Comment:
This is dangerous. You should be carefull while removing items from a list while iterating through it. It can cause a Concurrent Modification Exception.

Reply:
fixed. although, I can change it to work with BottomNavigationItems if you think that's a better solution, that way we can use the other get(Object). For now it goes with that solution as I don't have more time today